prov/sm2/src/sm2.h

```diff
@@ -102,6 +103,11 @@ enum {
 /* The SM2_UNEXP flag is used for unexpected receives. On the receiver, it
 indicates that the xfer_entry that the receiver is processing is actually in the
 xfer_ctx_pool and not in the receiver's freestack.
+
+For protocols that require delivery complete semantics (CMA and IPC), the
+receiver pops an xfer_entry from its own freestack, sets SM2_UNEXP flag together
+with SM2_RETURN and SM2_GENERATE_COMPLETION and sends it to the sender. The
```

Comment:
This is a little confusing. Are you saying that's for the unexpected path? Setting UNEXP with GENERATE_COMPLETION seems contradictory

Reply:
Updated comment